ui/ui-components/client/client.ts

```diff
@@ -190,20 +194,94 @@ export class Client {
       }
 
       if (error.response && error.response.data && error.response.data.error) {
+        if (this.redirectOnAccessError(error.response.data.error)) {
+          return new Promise<Response>(() => {});
```

Comment:
This stops the page from resolving anything else while the page just redirects to where it needs to go.

Reply:
I'm unsure if this is a memory leak since this promise is never resolved.

Reply:
should this just `throw` and let the `catch` on the `request` method handle it?

Reply:
Yes, I was also wondering if it would cause a leak. Basically, it stops the request from resolving so that it can redirect.

For example:

```ts
const onButtonClick = () => {
   setLoading(true)
    // This will below never resolve, so the page will stop being updated
   const response = await client.request('get', '/will-get-logged-out')
   setResponse(response)
   setLoading(false)
}
```

Reply:
I would suggest throwing as well to stop processing.  Throwing also stops subsequent requests, especially during page hydration